BASE/src/Microsoft.ApplicationInsights/Channel/Transmission.cs

```diff
@@ -133,6 +134,47 @@ public ICollection<ITelemetry> TelemetryItems
             get; private set;
         }
 
+        /// <summary>
+        /// Gets a value indicating whether flush is async.
+        /// </summary>
+        public bool HasFlushTask => this.flushTaskCompletionSource != null && !this.flushTaskCompletionSource.Task.IsCompleted;
+
+        /// <summary>
+        /// Gets Test to support async Flush.
+        /// </summary>
+        public Task<bool> GetFlushTask(CancellationToken cancellationToken)
+        {
+            if (this.flushTaskCompletionSource == null)
+            {
+                this.flushTaskCompletionSource = new TaskCompletionSource<bool>();
```

Comment:
I think if this is called by multiple threads you have a potential for race and double initialization + double registration in .Register.

Reply:
It could be problematic, for the first task that would get initialized. Unless it would be cancelled, it would never complete.

Reply:
This gets called only when client calls FlushAsync API and not meant to be called as standalone. Each transmission will have its own TaskCompletionSource. TelemetryChannel and Microsoft.ApplicationInsights are two different project, we have to make this method as public for TelemetryChannel to consume it.

Reply:
Editorbrowsable can be set to false, if this was not intended for consumers to use.